lib/datadog/core/environment/platform.rb

```diff
@@ -0,0 +1,38 @@
+# typed: false
+
+require 'datadog/core/environment/identity'
+
+module Datadog
+  module Core
+    module Environment
+      # For gathering information about the platform
+      module Platform
+        module_function
+
+        # @return [String] name of host; `uname -n`
+        def hostname
+          Identity.lang_version >= '2.2' ? Etc.uname[:nodename] : nil
+        end
+
+        # @return [String] name of kernel; `uname -s`
+        def kernel_name
+          Identity.lang_version >= '2.2' ? Etc.uname[:sysname] : Gem::Platform.local.os.capitalize
+        end
+
+        # @return [String] kernel release; `uname -r`
+        def kernel_release
```

Comment:
On linux it looks like JRuby behaves differently, actually matching `uname -r`.
```
$ docker run --rm -it jruby:9.2.8.0-jre ruby -e "require 'etc';puts Etc.uname"
{:sysname=>"Linux", :nodename=>"f7adb69905a6", :release=>"unknown", :version=>"5.10.104-linuxkit", :machine=>"amd64"}
$ docker run --rm -it jruby:9.2.8.0-jre uname -r
5.10.104-linuxkit
```

Reply:
I recommend checking out the whole `Etc.uname` data structure in a JRuby linux container and see the difference.